### ISSUE TITLE
Sluggable generate unique slug bug

### ISSUE THREAD

**Issue by hermann8u** [KnpLabs/DoctrineBehaviors#471]
Hi,

I have an issue with the v2.0.0-alpha4 with the generation of a unique Slug on the method Knp\DoctrineBehaviors\ORM\Sluggable\SluggableSubscriber::generateUniqueSlugFor.

This error occurs when I create my entity for the first time.

The issue is at line 84 because my slug is null at this moment and the interface define the getSlug method with on non nullable string as return type.

I defined my slug like this

 ```
    public function shouldGenerateUniqueSlugs(): bool
    {
        return true;
    }

    private function getRegenerateSlugOnUpdate(): bool
    {
        return false;
    }
```

## EDIT

If we look at the code of the *SluggableSubscriber*, we can see in the *processLifecycleEventArgs* that the method *generateSlug* is called. Because I put the result of getRegenerateSlugOnUpdate to false in my entity, the slug is never generated (see Knp\DoctrineBehaviors\Model\Sluggable\SluggableMethodsTrait:32 )

So finally, the bug is not related with uniqueness of the slug but with the getRegenerateSlugOnUpdate method when the result is set to false

**Comment by TomasVotruba** [comment]
Hi, thanks for repoting this.

Any idea how to fix it?

**Comment by hermann8u** [comment]
To me, the easiest way to fix it is to complete the first check I the generateSlug of the trait method. If the slug is null, we continue even if the 
getRegenerateSlugOnUpdate method return false.

I'll do a PR if you want in some days but for the moment it's Christmas time!

**Comment by TomasVotruba** [comment]
> I'll do a PR if you want in some days but for the moment it's Christmas time!

:+1:

**Comment by hermann8u** [comment]
This issue is fixed by the PR #472
